WooCommerce/Classes/ViewRelated/Blaze/BlazeEditAd/BlazeEditAdViewModel.swift

```diff
@@ -0,0 +1,204 @@
+import Foundation
+import Combine
+
+/// View model for `BlazeEditAdView`
+final class BlazeEditAdViewModel: ObservableObject {
+    typealias ImageState = EditableImageViewState
+
+    let siteID: Int64
+    private let adData: BlazeEditAdData
+
+    // Image selection
+    var onAddImage: ((MediaPickingSource) async -> MediaPickerImage?)?
+    @Published private(set) var imageState: ImageState
+    var image: MediaPickerImage? {
+        imageState.image
+    }
+
+    // Tagline
+    @Published var tagline: String = ""
+    var taglineFooterText: String {
+        taglineEmptyError ?? taglineLengthLimitLabel
+    }
+    private let taglineMaxLength = 32
+    @Published private var taglineRemainingLength: Int
+    private var taglineLengthLimitLabel: String {
+        String(format: Localization.taglineLengthLimit, taglineRemainingLength)
+    }
+    private var taglineEmptyError: String?
+
+    // Description
+    @Published var description: String = ""
+    var descriptionFooterText: String {
+        descriptionEmptyError ?? descriptionLengthLimitLabel
+    }
+    private let descriptionMaxLength = 140
+    @Published private var descriptionRemainingLength: Int
+    private var descriptionLengthLimitLabel: String {
+        String(format: Localization.descriptionLengthLimit, descriptionRemainingLength)
```

Comment:
Similar comment as above about pluralizing the text based on the remaining length.

Reply:
Done as part of https://github.com/woocommerce/woocommerce-ios/commit/72a88cec7152cf4a946012e66615e9e2b73df0ee